src/components/TableEditorDialog.vue

```diff
@@ -0,0 +1,222 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+<script>
+import { NcButton, NcModal, NcNoteCard } from '@nextcloud/vue'
+import { defineComponent } from 'vue'
+import { t } from '@nextcloud/l10n'
+
+export default defineComponent({
+	name: 'TableEditorDialog',
+	components: {
+		NcButton,
+		NcModal,
+		NcNoteCard,
+	},
+	props: {
+		initialMarkdown: {
+			type: String,
+			default: '',
+		},
+	},
+	emits: ['cancel', 'submit'],
+	data() {
+		return {
+			show: true,
+			editor: null,
+			isLoading: true,
+			error: null,
+			currentMarkdown: this.initialMarkdown || this.getDefaultTable(),
+		}
+	},
+	computed: {
+		isEditing() {
+			return Boolean(this.initialMarkdown)
+		},
+	},
+	async mounted() {
+		await this.$nextTick()
+		await this.initializeEditor()
+	},
+	beforeUnmount() {
+		this.destroyEditor()
+	},
+	methods: {
+		t,
+		async initializeEditor() {
+			try {
+				// Check if Text app is available
+				if (!window.OCA?.Text) {
+					this.error = t('whiteboard', 'Nextcloud Text app is not available. Please install and enable it.')
+					this.isLoading = false
+					return
+				}
+
+				const editorContainer = this.$refs.editorContainer
+				if (!editorContainer) {
+					this.error = t('whiteboard', 'Editor container not found')
+					this.isLoading = false
+					return
+				}
+
+				// Create the Text editor instance with callbacks
+				this.editor = await window.OCA.Text.createEditor({
+					el: editorContainer,
```

Comment:
If we only want to use tables, we could think about offering a way in the text app to only allow one table node in the document, but i think using this as a general rich text node is nice as well.